### PR TITLE
Add internal ip whitelist option to healthcheck to be appended to the default

### PR DESCRIPTION
**Pull request by wickstargazer**
Added WhitelistedHealthCheckCIDR option to main.go skipper.go and kube.go as string array
append WhitelistedHealthCheckCIDR to internalIps in kube.go
tested on docker kubernetes, healthcheck working fine

### CODE REVIEW

**Comment by hjacobs** [comment]
@wickstargazer can you describe your use case / usage scenario, i.e. why do you propose this change?

**Comment by szuecs** [comment]
@hjacobs https://github.com/zalando/skipper/issues/644#issuecomment-389468511 it's serious and some people already asked for it.
@wickstargazer thanks for the PR, I will check it today.

File: dataclients/kubernetes/kube.go [hunk 1]
```diff
@@ -171,6 +174,17 @@ func New(o Options) (*Client, error) {
 		signal.Notify(sigs, syscall.SIGTERM)
 	}
 
+	log.Infof("extra whitelist urls: %s", o.WhitelistedHealthCheckCIDR)
```
Comment (szuecs) [inline dataclients/kubernetes/kube.go]: %s -> %v

File: dataclients/kubernetes/kube.go [hunk 2]
```diff
@@ -171,6 +174,17 @@ func New(o Options) (*Client, error) {
...
+	if len(o.WhitelistedHealthCheckCIDR) > 0 {
+		t := o.WhitelistedHealthCheckCIDR
+		whitelistCIDS := make([]interface{}, len(t))
+		for i, v := range t {
+			whitelistCIDS[i] = v
+		}
+		internalIPs = append(internalIPs , whitelistCIDS...);
+		log.Infof("new internal ips are: %s", internalIPs)
```
Comment (szuecs) [inline dataclients/kubernetes/kube.go]: %s -> %v

File: dataclients/kubernetes/kube.go [hunk 3]
```diff
@@ -171,6 +174,17 @@ func New(o Options) (*Client, error) {
 		signal.Notify(sigs, syscall.SIGTERM)
 	}
 
+	log.Infof("extra whitelist urls: %s", o.WhitelistedHealthCheckCIDR)
+	if len(o.WhitelistedHealthCheckCIDR) > 0 {
+		t := o.WhitelistedHealthCheckCIDR
+		whitelistCIDS := make([]interface{}, len(t))
```
Comment (szuecs) [inline dataclients/kubernetes/kube.go]: whitelistCIDS -> whitelistCIDRs

File: dataclients/kubernetes/kube.go [hunk 4]
```diff
@@ -171,6 +174,17 @@ func New(o Options) (*Client, error) {
 		signal.Notify(sigs, syscall.SIGTERM)
 	}
 
+	log.Infof("extra whitelist urls: %s", o.WhitelistedHealthCheckCIDR)
+	if len(o.WhitelistedHealthCheckCIDR) > 0 {
+		t := o.WhitelistedHealthCheckCIDR
```
Comment (szuecs) [inline dataclients/kubernetes/kube.go]: drop the t and always use the options reference please

File: skipper.go [hunk 5]
```diff
@@ -415,6 +418,8 @@ type Options struct {
 
 	// MaxAuditBody sets the maximum read size of the body read by the audit log filter
 	MaxAuditBody int
+
+
```
Comment (szuecs) [inline skipper.go]: please cleanup empty lines

File: skipper.go [hunk 6]
```diff
@@ -46,6 +46,9 @@ const DefaultPluginDir = "./plugins"
 // Options to start skipper.
 type Options struct {
 
+	//WhitelistedHealthcheckCIDR appends the whitelisted IP Range to the inernalIPS range for healthcheck purposes
```
Comment (szuecs) [inline skipper.go]: Please add a whitesapce before comment and first word:

    // WhitelistedHealthcheckCID

File: skipper.go [hunk 7]
```diff
@@ -476,7 +481,7 @@ func createDataClients(o Options, auth innkeeper.Authentication) ([]routing.Data
 
 		clients = append(clients, etcdClient)
 	}
-
+    log.Infof("extra whitelist urls: %s", o.WhitelistedHealthCheckCIDR)
```
Comment (szuecs) [inline skipper.go]: delete this log line, you already log this in the dataclient

**Comment by szuecs** [comment]
@wickstargazer I added some comments and please squash the commit into one. Thanks!

**Comment by szuecs** [comment]
@wickstargazer sure I will notify in slack, twitter and https://discuss.kubernetes.io/c/announcements :)

I guess I should also post to our google group, which I don't do so often. :(

**Comment by wickstargazer** [comment]
Ok squashed the commits and fixed the comments

**Comment by szuecs** [comment]
:+1:

File: dataclients/kubernetes/kube.go [hunk 8]
```diff
@@ -171,6 +174,15 @@ func New(o Options) (*Client, error) {
 		signal.Notify(sigs, syscall.SIGTERM)
 	}
 
+	if len(o.WhitelistedHealthCheckCIDR) > 0 {
+		whitelistCIDRS := make([]interface{}, len(o.WhitelistedHealthCheckCIDR))
+		for i, v := range t {
```
Comment (szuecs) [inline dataclients/kubernetes/kube.go]: @wickstargazer `t` is not defined here, please check the travis build

**Comment by wickstargazer** [comment]
Oops will push again dint change all the t

File: skipper.go [hunk 9]
```diff
@@ -476,7 +479,7 @@ func createDataClients(o Options, auth innkeeper.Authentication) ([]routing.Data
 
 		clients = append(clients, etcdClient)
 	}
-
+
```
Comment (szuecs) [inline skipper.go]: please remove this whitespace.
You can also run "gofmt -sw" on this file

Comment (szuecs) [reply skipper.go]: see https://travis-ci.org/zalando/skipper/builds/385732378?utm_source=github_status&utm_medium=notification

Comment (aryszka) [reply skipper.go]: or `make fmt`

**Comment by wickstargazer** [comment]
Hey guys been busy

I will fmt the file toMorrow or is it done?

**Comment by aryszka** [comment]
@wickstargazer Thanks a lot for this PR! :smile: 

(I just went ahead and formatted and merged master)

**Pull request merged**